In [1]:
import pandas as pd
from neuronunit.optimization.optimization_management import public_opt
from sciunit.scores import RelativeDifferenceScore, ZScore
from sciunit.utils import config_set, config_get
config_set('PREVALIDATE', False)
from neuronunit.allenapi.allen_data_driven import make_allen_hard_coded_limited, make_allen_hard_coded_complete
import warnings
from neuronunit.optimization.model_parameters import BPO_PARAMS
warnings.filterwarnings("ignore")
[allen_suite_623960880,_,_,_,_,df] = make_allen_hard_coded_complete()

/home/user/anaconda3/lib/python3.7/site-packages/cerberus/validator.py:1609: UserWarning: No validation schema is defined for the arguments of rule 'not_zero_obs_zscore'
  "'%s'" % method_name.split('_', 2)[-1]
test


# In this notebook
we fit Izhikevich and Adaptive Exponential Integrate and fire models to electrophysiology observations 
from the Allen Brain, using BluePyOpt and NeuronUnit.

* Immediately Below 
is a list of some of the observations we will use:

In [2]:
df

TimeConstantTest       name RestingPotentialTest  \
623960880          23.8 ms  623960880             -65.1 mV   
623893177          27.8 ms  623893177             -77.0 mV   
471819401          13.8 ms  471819401             -77.5 mV   
482493761          24.4 ms  482493761             -71.6 mV   

          InputResistanceTest RheobaseTest  
623960880       241.0 megaohm      70.0 pA  
623893177       136.0 megaohm     190.0 pA  
471819401       132.0 megaohm     190.0 pA  
482493761       132.0 megaohm      70.0 pA

Allen Specimen ID: 471819401
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model AdExp

In [ ]:
_,_,_,a_cells = make_allen_hard_coded_limited()
opt,obs_preds,df = public_opt(
    a_cells['471819401'],
    BPO_PARAMS,
    '471819401',
    "ADEXP",
    100,
    300,
    "IBEA",
    score_type=RelativeDifferenceScore
    )


gen	nevals	avg    	std    	min    	max 
1  	100   	1528.25	1520.98	12.4622	4000


2  	45    	1490.93	1481.37	12.4622	4000
3  	6     	426.875	489.645	12.4622	1010.18
4  	11    	571.687	493.039	12.4622	1009.25
5  	10    	668.375	470.728	12.4622	1009.62
6  	9     	353.069	470.693	12.4622	1008.29
7  	12    	594.752	489.142	7.58842	1009.33
8  	12    	278.79 	440.966	7.58842	1009.46
9  	8     	253.804	427.517	7.58842	1008.26
10 	9     	325.699	463.286	7.58842	1008.23
11 	6     	277.489	441.083	7.58842	1008.23
12 	6     	422.241	490.839	7.58842	1007.72
13 	9     	251.46 	427.893	7.46956	1005.43
14 	8     	445.977	494.475	7.46956	1008.04
15 	9     	300.464	452.9  	7.12105	1005.77
16 	6     	519.263	497.486	7.12105	1006.76
17 	5     	81.901 	259.394	7.12105	1006.77
18 	8     	81.0754	260.121	7.10375	1007.51
19 	5     	105.818	296.049	7.10375	1006.52
20 	7     	153.991	352.285	7.10375	1006.08
21 	13    	154.692	352.296	7.10375	1007.05
22 	9     	275.449	441.729	7.10375	1007.06
23 	2     	421.713	490.822	7.10375	1005.66
24 	5     	348.357	473.026	7.10375	1005.54
25 	9     	105

In [ ]:
df

# Solution Model Parameters:


In [ ]:
opt.attrs_to_params()
opt.params

Allen Specimen ID: 471819401
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model Izhikevich

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['471819401'],
    BPO_PARAMS,
    'fi_curve',
    "IZHI",
    100,
    360,
    "IBEA",
    )

In [ ]:
df

# Solution Parameters:

In [ ]:
opt.attrs_to_params()
opt.params

# Just FI curve on AdExp model
Allen Specimin Id: 471819401

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['fi_curve'],
    BPO_PARAMS,
    'fi_curve',
    "ADEXP",
    100,
    100,
    "IBEA"
    )

In [ ]:
df


In [ ]:
opt.attrs

Allen Specimen ID: 623960880
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model Izhikevich

In [ ]:

opt,obs_preds,df = public_opt(
    allen_suite_623960880.tests,
    BPO_PARAMS,
    allen_suite_623960880.name,
    "IZHI",
    100,
    100,
    "IBEA"
    )

In [ ]:
df

# Solution parameters

In [ ]:
opt.attrs_to_params()
opt.params